### DISCUSSION TITLE
Can't run a second Postgrest instance

### DISCUSSION THREAD

**Discussion by Inrap** [PostgREST/postgrest#2938] [Q&A]
windows server 2012, PostgreSQL14, PostGIS 3.1, PostgREST 11.2
PostgreSQL and PostgREST are on the same server. 

-----
I want to access an other database from a second `PostgREST` instance.

Here is the config file of the first instance of `PostgREST` running on the database "activite" : 

    db-uri = "postgres://postgrestuser:xxxx@localhost:5432/activite"
    db-schemas = "activite,public"
    db-anon-role = "postgrestuser"
    db-pool-max-lifetime = 60

The API is running fine.

Here is the second config file for the second instance that I want to create on the database "activite_test" : 

    db-uri = "postgres://postgrestuser:xxxx@localhost:5432/activite_test"
    db-schemas = "activite,public"
    db-anon-role = "postgrestuser"
    db-pool-max-lifetime = 60
    server-port = 3001

In a commande line : `postgrest postgrest_3001.conf` returns two errors : 

- An error occured when loading the schema cache
- GDAL AppDefined only read-only mode is supported for /vsicurl

As you can see, the user is identical and databases are the same (schemas, tables,...)
I checked the rights in both of them : as far as i know they are identical as well.
Thus,I can't understand why GDAL is involved.

Thanks for any idea !

**Comment by steve-chavez** [comment]
> In a commande line : postgrest postgrest_3001.conf returns two errors :
> An error occured when loading the schema cache
> GDAL AppDefined only read-only mode is supported for /vsicurl

Can you share the full output as shown in the terminal?

**Comment by Inrap** [comment]
Here is the first attempt (it keeps trying to launch the instance)

![image](https://github.com/PostgREST/postgrest/assets/32261793/44c9721b-4783-4b1f-8848-a195c5a40906)

**Reply by wolfgangwalther** [reply]
Do you have any foreign tables in your schema?

Are you using PostGIS?

`/vsicurl` seems to be related to this: https://gdal.org/user/virtual_file_systems.html#vsicurl-http-https-ftp-files-random-access - this seems to be about file system access on windows. Are you refering to an external file via FDW or PostGIS anywhere?

**Reply by wolfgangwalther** [reply]
Also see this issue: https://github.com/pramsey/pgsql-ogr-fdw/issues/219

This does not seem to be a problem with PostgREST.

**Comment by Inrap** [comment]
> Are you using PostGIS?

I am using PostGIS 3.1 (as I mentionned at the top of the first message).

> Do you have any foreign tables in your schema?

**EDIT** : There is a FDW on the activite_test database.

The first instance of PostgREST is running fine.
Even if I stop it, I can't create an other instance on port 3001 (or 3000!) : the same error message is sent. Weird...
The schema cache seems to be involved.

Thanks for the links.

**Comment by Inrap** [comment]
A few weeks ago, I tried [this sort of FDW](https://www.crunchydata.com/blog/remote-access-anything-from-postgres).
It uses `vsicurl`. I forgot it because it was a test and it worked more or less as expected.
Since all this is in a test server, I delete this FDW and now I can create a second instance of `PostgREST` on port 3001.

As to why an FDW prevents an instance from being launched, that's beyond my comprehension.
